drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -53,6 +54,28 @@ local function device_removed(driver, device)
   log.info("device removed")
 end
 
+local function do_configure(driver, device)
+  -- New profiles need to be added for devices that have more switch endpoints
+  local MAX_MULTI_SWITCH_EPS = 7
+  -- Note: This profile switching is needed because of shortcoming in the generic fingerprints
+  -- where devices with multiple endpoints with the same device type cannot be detected
+  local switch_eps = device:get_endpoints(clusters.OnOff.ID)
+  local num_switch_eps = #switch_eps
+  table.sort(switch_eps)
+  --Default MCD switch handling depends on consecutive endpoint numbering
+  if num_switch_eps == switch_eps[num_switch_eps] then
+    if num_switch_eps > 1 then
+      device:try_update_metadata({profile = string.format("switch-%d", math.min(num_switch_eps, MAX_MULTI_SWITCH_EPS))})
+    end
+    if num_switch_eps > MAX_MULTI_SWITCH_EPS then
+      error(string.format(
+        "Matter multi switch device will not function. Profile doesn't exist with %d components",
```

Comment:
I think the message should indicate that it's only endpoints beyond X that won't work

Reply:
Well it would actually affect the main components endpoint, so that endpoint also wouldn't work.

Reply:
what do you mean?

Reply:
> What will happen is that reports from the endpoints higher than the max we support will be mapped to the main component.

This means manual changes to those switches will be displayed on the main state. So not only will the higher endpoints not work, their presence will make the main component change state when it shouldn't in some cases.

Reply:
Ah right. Because even if `endpoint_to_component` returns `nil` we throw in a `main`. Ran into some annoyances with that earlier this week as well.